posts/2023-11-05-telegram-bot-for-file-sharing-using-quarkus.adoc

```diff
@@ -0,0 +1,546 @@
+      - 9001:9001
+    volumes:
+      - ~/minio/data:/data
+    environment:
+      - MINIO_ROOT_USER=testtest
+      - MINIO_ROOT_PASSWORD=testtest
+    command: server /data --console-address ":9001"
+    networks:
+      - nw
+
+networks:
+  nw:
+----
+
+After we run docker-compose our backend application is ready to up.
+Being in `/backend` directory run
+```bash
+quarkus dev
+```
+
+
+== Frontend
+For the frontend application, we will use https://docs.quarkiverse.io/quarkus-quinoa/dev/[Quarkus Quinoa project] using React with TypeScript.
+All possible framework combinations you can see on https://docs.quarkiverse.io/quarkus-quinoa/dev/web-frameworks.html.
+
+Being in the root of our repository execute
+```bash
+quarkus create app frontend -x=io.quarkiverse.quinoa:quarkus-quinoa --gradle
+```
+
+Inside our generated application we can see the directory  `src/main/webui` - it’s a root for our NodeJS application with `package.json` file. Let's remove the default implementation by removing the whole directory  `src/main/webui` and being in a  `frontend/src` execute
+```bash
+npx create-react-app webui --template typescript
+```
+(it requires an already installed `node` and `npx`).
+
+In directory `src/main/webui` we can see the new generated project.
+
+As usual, let's add `bootstrap`. Being in `/frontend/src/main/webui` execute
+```bash
+npm install bootstrap
+```
+
+The last thing we need is to change the `webui/src/App.tsx` file. We have to add one button and function that sends an HTTP request to our backend application for `onClick()` event. Also we will retrieve the chatId variable from the query parameter.
+[source,javascript]
+----
+import React from 'react';
+import logo from './logo.svg';
+import './App.css';
+import '../node_modules/bootstrap/dist/css/bootstrap.css';
+
+function App() {
+  const queryParams = new URLSearchParams(window.location.search);
+  const chatId = queryParams.get('chatId');
+
+  function download() {
+    window.location.href = `http://127.0.0.1:8091/api/v1/file-sharing-bot/download?chatId=${chatId}`;
+  }
+
+  return (
+      <div className="App">
+        <h1 className="display-1"> Click on the button to download</h1>
+        <button className="btn btn-primary" onClick={() => download()}>Download</button>
+      </div>
+  );
+}
+
+export default App;
+
+----
+
+Yes, hardcode it (3).
+
+And just in case change the default port for our application by adding the property `quarkus.http.port=3006` to `application.properties`
+
+Now we can build our application by
+```bash
+quarkus build
+```
+and run it
+```bash
+java -jar build/quarkus-app/quarkus-run.jar
+```
+
+Let's check that our app returns at least something. Open http://localhost:3006/
+
+image::frontend.png[,600]
+
+
+== User case
+
+What do we have now?
+
+1. Backend application
+2. Frontend application
+3. MongoDB and MinIO in docker
+
+And so, the final test case.
+
+1) Open the bot
+
+image::step1.png[,600]
+2) Click on "Start" button
```

Comment:
You mean "Send the /start command", right?

Reply:
if we open the bot in the first time we will see only the "Start" button, which is sending "/start" command. Changed to "Click on the "Start" button, which is sending the `/start` command."